Bitbucket.Authentication/Authority.cs

```diff
@@ -87,59 +112,16 @@ public string UserUrl
             }
             else
             {
-                // use the provided username and password and attempt a Basic Auth request to a known
-                // REST API resource.
-                Token token = null;
-                using (HttpClientHandler handler = targetUri.HttpClientHandler)
+                BasicAuthAuthenticator basicauth = new BasicAuthAuthenticator();
+                try
                 {
-                    using (HttpClient httpClient = new HttpClient(handler) { Timeout = TimeSpan.FromMilliseconds(RequestTimeout) })
```

Comment:
Refactored out the call the /user REST api because it is used in 3 places

1. check username/password basic auth credentials
1. validate any credentials
1. retrieve the username from Bitbucket